src/AtomShell/install.jl

```diff
@@ -42,11 +42,31 @@ function install()
     end
 
     if Sys.iswindows()
-      arch = Int == Int64 ? "x64" : "ia32"
-      file = "electron-v$version-win32-$arch.zip"
-      download("https://github.com/electron/electron/releases/download/v$version/$file")
-      run(`7z x $file -oatom`)
-      rm(file)
+       arch = Int == Int64 ? "x64" : "ia32"
+       file = "electron-v$version-win32-$arch.zip"
+       download("https://github.com/electron/electron/releases/download/v$version/$file")
+       zipper = joinpath(Base.Sys.BINDIR, "7z.exe")
+       if !isfile(zipper)
+           #=
+           This is likely built with cygwin, which includes its own version of 7z.
+           But if we unzip with cmd = Cmd(`$(fwhich("bash.exe")) 7z x $file -oatom`)
+           The resulting files would not be windows executable.
+           So we want the 7z.exe included with a binary download of Julia.
+           The PATH environment variable likely includes to the locally built
+           Julia, so instead we look in the default Julia binary location and
+           pick the latest version.
+           =#
+           juliafolders = filter(readdir(ENV["LOCALAPPDATA"])) do f
+                                     startswith(f, "Julia-")
+              end
+           juliaversions = VersionNumber.([replace(f, "Julia-" => "") for f in juliafolders])
+           i = findlast(isequal(maximum(juliaversions)), juliaversions)
+           zipper = joinpath(ENV["LOCALAPPDATA"], juliafolders[i], "bin", "7z.exe")
+           isfile(zipper) || error("could not find $zipper. Try also installing a binary version of Julia.")
```

Comment:
The idea of looking across julia installations for dependencies seems a bit dicey to me. :( If there ends up being a version compatibility issue that's fixed, we might have problems if we pick up old/bad versions?

Can we maybe just ask users to install `7Zip` manually if you're going to use a native-built julia, and then fallback to looking for `7z` in `C:\Program Files\7-Zip\`?

Reply:
> ask users to install 7Zip manually if you're going to use a native-built julia

We could ask users to put their 7Zip installation in a folder of the PATH environment variable. Windows user in general would hate that, but not so much the ones who use cygwin. 

If using the "native-built" julia, as in built via a cygwin terminal, 7z is already installed as part of cygwin. My original intention was to just use that. From the referred issue:
```
          # This is likely built with cygwin; run the version of 7z included with it.
          cmd = Cmd(`$(fwhich("bash.exe")) 7z x $file -oatom`)
```
However, I discovered that the extracted files were not runnable from windows. Hence the change to using the latest binary Julia install. I do not know which features of the files makes them incompatible with Windows.

For info, Julia 1.0.3 has since been released, but the bug concerning paths to standard libraries has not been fixed.  I believe a good part of the contributing community is using Windows these days, which gives some hope of better binaries going forward. The 7z problem is perhaps affecting hundreds of packages, and getting most of them do this hack is not realistic. 

My "solution" for other packages is now to do the installation from the downloaded binary. But this seems to trigger recompilation of every other package used during the build process. For deep dependency trees it can take a lot of time to get it all working simultaneously.

To conclude, I think this PR is an ugly hack, but I would still call it an improvement. I do not yet have the competence to identify the underlying issue properly.